nats/src/main.rs

```diff
@@ -216,7 +262,19 @@ impl ProviderHandler for NatsMessagingProvider {
     /// If the link is allowed, return true, otherwise return false to deny the link.
     #[instrument(level = "debug", skip(self, ld), fields(actor_id = %ld.actor_id))]
     async fn put_link(&self, ld: &LinkDefinition) -> RpcResult<bool> {
-        let config = ConnectionConfig::new_from(&ld.values)?;
+        // If the link definition values are empty, use the default connection configuration
+        let config = if ld.values.is_empty() {
+            self.default_config.clone()
+        } else {
+            // create a config from the supplied values and merge that with the existing default
+            match ConnectionConfig::new_from(&ld.values) {
+                Ok(cc) => self.default_config.merge(&cc),
```

Comment:
Do we need both `new_from` and `merge`? They're both only referenced once -- here, in the same spot -- so I think the logic from one could move into the other?

Reply:
I looked at this a couple of times and I don't think merging them (see what I did there?) will add that much , other than making the "merge_from" function longer than a page.